### ISSUE TITLE
SLSv4: Issue with parameters for plugin defined variables sources

### ISSUE THREAD

**Issue by alexandre-walzberg** [serverless/serverless#12816]
### Issue description

Upgrading to serverless v4 makes problems with plugin previously working in serverless v3.
Documentation about plugins for v4 did not change, and there should not be any breaking changes as said in the release note.

However, my plugin does not receive any parameter in the resolve function of a configurationVariablesSources defined by the plugin. Logs showed that all parameters are given into the "address" parameter and not into the "params" parameter. "params" remains undefined.
https://www.serverless.com/framework/docs/guides/plugins/custom-variables

### Context

Here is the output and error I get : 
```
⠏ InitializingPLUGIN lOGS:   "hello-dev", "dev", ${opt:stage} 
address:   "hello-dev", "dev", ${opt:stage} 
params:  undefined

✖ Failed to resolve variable ' "hello-dev", "dev", ${opt:stage} ' with resolver 'cwdashboard' and provider 'cwdashboard': TypeError: Cannot read properties of undefined (reading '0')
Error: Failed to resolve variable ' "hello-dev", "dev", ${opt:stage} ' with resolver 'cwdashboard' and provider 'cwdashboard': TypeError: Cannot read properties of undefined (reading '0')
    at #resolveKey (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:49101)
    at async _ResolverManager.resolve (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:54420)
    at async #handlePlaceholderNode (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:53234)
    at async file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:51233
    at async processNodeAndHandleCompletion (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:42013)
```
note the plugin output : 
**address:   "hello-dev", "dev", ${opt:stage} 
params:  undefined**

plugin.js
```
'use strict';
 
class CustomDashboard {
  constructor(serverless, cliOptions, { log }) {
    this.serverless = serverless;
    this.configurationVariablesSources = {
        cwdashboard: {
          async resolve({address, params}) {
            console.log("PLUGIN lOGS: ", address)
            console.log("address: ", address)
            console.log("params: ", params)
            return {value: JSON.stringify(params[0]).replaceAll(params[1], params[2])}
          }
        }
      };
  }
}

module.exports = CustomDashboard;
```
serverless.yml : 
```
service: service

plugins:
  - ./plugin.js

provider:
  name: aws
  region: eu-west-1
  runtime: nodejs18.x
  stage: ${opt:stage, 'dev'}

resources:
  dashboard:
    Type: AWS::CloudWatch::Dashboard
    Properties:
      DashboardName: ${opt:stage}-vod-dashboard
      DashboardBody: ${cwdashboard( "hello-dev", "dev", ${opt:stage} )}
```

**Comment by czubocha** [comment]
Hey @alexandre-walzberg, thanks for reporting. We're looking into it.

**Comment by czubocha** [comment]
Hey @alexandre-walzberg, this has been addressed in version 4.4.3, which is now live. Feel free to try it out and let us know if you run into any issues.
